source/internal/_path.js

```diff
@@ -0,0 +1,43 @@
+import _isInteger from './_isInteger.js';
+import _nth from './_nth.js';
+
+/**
+ * Retrieves the value at a given path. The nodes of the path can be arbitrary strings or non-negative integers.
+ * For anything else, the value is unspecified. Integer paths are meant to index arrays, strings are meant for objects.
+ *
+ * @func
+ * @category Object
+ * @typedefn Idx = String | Int | Symbol
+ * @sig ([Idx], {a}) -> a | Undefined
+ * @sig Idx = String | NonNegativeInt
+ * @param {Array} path The path to use.
+ * @param {Object} obj The object or array to retrieve the nested property from.
+ * @return {*} The data at `path`.
+ * @example
+ *
+ *      path(['a', 'b'], {a: {b: 2}}); //=> 2
+ *      path(['a', 'b'], {c: {b: 2}}); //=> undefined
+ *      path(['a', 'b', 0], {a: {b: [1, 2, 3]}}); //=> 1
+ *      path(['a', 'b', -2], {a: {b: [1, 2, 3]}}); //=> 2
+ *      path([2], {'2': 2}); //=> 2
+ *      path([-2], {'-2': 'a'}); //=> undefined
+ */
+
+function path(pathAr, obj) {
+  let val = obj;
```

Comment:
Let's just stick with `var` for now, no `let` or `const` except in comments.

Reply:
> However I did discover that clone.js uses let instead of var, should we change that too or is there a reason for that?

I noticed that haha, snuck in at some point. No reason, we can update it as part of this PR or I can submit one later for the clone stuff.

The use of var is historically to support es3. Once we get to 1.0 we can refactor to use let/const